e2e/cypress/integration/dashboard-transloadit.spec.ts

```diff
@@ -50,70 +44,81 @@ describe('Dashboard with Transloadit', () => {
     })
   })
 
-  it('should emit one assembly-cancelled event when cancelled', FLAKY, () => {
+  it('should emit one assembly-cancelled event when cancelled', () => {
     const spy = cy.spy()
 
-    cy.get('@file-input').selectFile(['cypress/fixtures/images/cat.jpg', 'cypress/fixtures/images/traffic.jpg'], { force:true })
-    cy.get('.uppy-StatusBar-actionBtn--upload').click()
-
-    cy.intercept({
-      method: 'GET',
-      url: '/assemblies/*',
-    }).as('assemblyPolling')
-    cy.intercept(
-      { method: 'PATCH', pathname: '/files/*', times: 1 },
-      { statusCode: 204, body: {} },
-    )
-    cy.intercept(
-      { method: 'DELETE', pathname: '/resumable/files/*', times: 2 },
-      { statusCode: 204, body: {} },
-    ).as('fileDeletion')
-    cy.intercept(
-      { method: 'DELETE', pathname: '/assemblies/*', times: 1 },
-    ).as('assemblyDeletion')
-    cy.wait('@assemblyPolling')
     cy.window().then(({ uppy }) => {
       uppy.on('transloadit:assembly-cancelled', spy)
-    })
-    cy.get('button[data-cy=cancel]').click()
 
-    cy.wait('@assemblyDeletion').then(() => {
-      expect(spy).to.be.calledOnce
+      cy.get('@file-input').selectFile(['cypress/fixtures/images/cat.jpg', 'cypress/fixtures/images/traffic.jpg'], { force:true })
+      cy.get('.uppy-StatusBar-actionBtn--upload').click()
+
+      cy.intercept({
+        method: 'GET',
+        url: '/assemblies/*',
+      }).as('assemblyPolling')
+      cy.intercept(
+        { method: 'PATCH', pathname: '/files/*', times: 1 },
+        { statusCode: 204, body: {} },
+      )
+      cy.intercept(
+        { method: 'DELETE', pathname: '/resumable/files/*', times: 2 },
+        { statusCode: 204, body: {} },
+      ).as('fileDeletion')
+      cy.intercept(
+        { method: 'DELETE', pathname: '/assemblies/*', times: 1 },
+      ).as('assemblyDeletion')
+
+      cy.wait('@assemblyPolling').then(() => {
+        cy.get('button[data-cy=cancel]').click()
+
+        cy.wait('@assemblyDeletion').then(() => {
+          // Unfortunately, waiting on a network request somehow often results in a race condition.
+          // We just want to know wether this is called or not, so waiting for 2 sec to be sure.
+          // eslint-disable-next-line cypress/no-unnecessary-waiting
+          cy.wait(2000).then(() => {
```

Comment:
i guess this didn't help after all. Maybe this is really a bug in Uppy, where it sometimes doesn't emit `transloadit:assembly-cancelled`?